### PR TITLE
fixed route index, UUID, and RouteOptions assignment across types

### PR DESCRIPTION
**Pull request by LukasPaczos**
Closes https://github.com/mapbox/mapbox-java/issues/1292.

Additionally fixes an issue where `routes` provided `DirectionsResponse#builder` had to be a mutable list. Now, we're working on a copy.

### CODE REVIEW

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsResponse.java [hunk 1]
```diff
@@ -156,24 +161,71 @@ public static DirectionsResponse fromJson(@NonNull String json) {
    *   method
    * @see RouteOptions#fromUrl(java.net.URL)
    * @see RouteOptions#fromJson(String)
+   * @deprecated use {@link #fromJson(String, RouteOptions)} instead
    */
+  @Deprecated
   public static DirectionsResponse fromJson(
     @NonNull String json, @Nullable RouteOptions routeOptions, @Nullable String requestUuid) {
```
Comment (RingerJK) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsResponse.java:168]: nit
```suggestion
    @NonNull String json, @Nullable RouteOptions routeOptions, @Nullable String requestUuid
  ) {
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsResponse.java [hunk 2]
```diff
@@ -156,24 +161,71 @@ public static DirectionsResponse fromJson(@NonNull String json) {
...
+   *
+   * @param json         a formatted valid JSON string defining a GeoJson Directions Route
+   * @param routeOptions options that were used during the original route request
+   * @return a new instance of this class defined by the values passed inside this static factory
+   *   method
+   * @see RouteOptions#fromUrl(java.net.URL)
+   * @see RouteOptions#fromJson(String)
+   */
+  public static DirectionsResponse fromJson(
+    @NonNull String json, @NonNull RouteOptions routeOptions) {
```
Comment (RingerJK) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsResponse.java:199]: nit
```suggestion
    @NonNull String json, @NonNull RouteOptions routeOptions
  ) {
```

File: services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java [hunk 3]
```diff
@@ -339,7 +341,7 @@ public static Builder builder() {
      * @param usePost true to use POST method or false and null for GET method
      * @return this builder for chaining options together
      */
-    public abstract Builder usePostMethod(@Nullable  Boolean usePost);
+    public abstract Builder usePostMethod(@Nullable Boolean usePost);
```
Comment (RingerJK) [inline services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java:344]: could you cut a ticket to add a java linter to mapbox-java library

**Review by RingerJK — APPROVED** [review]
LGTM

File: services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java (hunk 3 above)
Comment (LukasPaczos) [reply services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java:344]: Checkstyle is configured in https://github.com/mapbox/mapbox-java/tree/main/config/checkstyle, looks like it's less strict than ktlint that we use in other projects though. Feel free to make any changes directly in there if you think that we need them.

**Pull request merged**